ethers-core/src/macros/ethers_crate.rs

```diff
@@ -1,123 +1,310 @@
 
-/// Convenience function to turn the `ethers_core` name in `ETHERS_CRATE` into a `Path`
-pub fn ethers_core_crate() -> Path {
-    syn::parse_str(ETHERS_CRATES.0).expect("valid path; qed")
+/// Returns the `core` crate's [`Path`][syn::Path].
+#[inline]
+pub fn ethers_core_crate() -> syn::Path {
+    get_crate_path(EthersCrate::EthersCore)
 }
-/// Convenience function to turn the `ethers_contract` name in `ETHERS_CRATE` into an `Path`
-pub fn ethers_contract_crate() -> Path {
-    syn::parse_str(ETHERS_CRATES.1).expect("valid path; qed")
+
+/// Returns the `contract` crate's [`Path`][syn::Path].
+#[inline]
+pub fn ethers_contract_crate() -> syn::Path {
+    get_crate_path(EthersCrate::EthersContract)
 }
-pub fn ethers_providers_crate() -> Path {
-    syn::parse_str(ETHERS_CRATES.2).expect("valid path; qed")
+
+/// Returns the `providers` crate's [`Path`][syn::Path].
+#[inline]
+pub fn ethers_providers_crate() -> syn::Path {
+    get_crate_path(EthersCrate::EthersProviders)
 }
 
-/// The crates name to use when deriving macros: (`core`, `contract`)
-///
-/// We try to determine which crate ident to use based on the dependencies of
-/// the project in which the macro is used. This is useful because the macros,
-/// like `EthEvent` are provided by the `ethers-contract` crate which depends on
-/// `ethers_core`. Most commonly `ethers` will be used as dependency which
-/// reexports all the different crates, essentially `ethers::core` is
-/// `ethers_core` So depending on the dependency used `ethers` ors `ethers_core
-/// | ethers_contract`, we need to use the fitting crate ident when expand the
-/// macros This will attempt to parse the current `Cargo.toml` and check the
-/// ethers related dependencies.
-///
-/// This determines
-///   - `ethers_*` idents if `ethers-core`, `ethers-contract`, `ethers-providers`  are present in
-///     the manifest or the `ethers` is _not_ present
-///   - `ethers::*` otherwise
+/// Returns an [`EthersCrate`]'s [`Path`][syn::Path] in the current project.
+#[inline(always)]
+pub fn get_crate_path(krate: EthersCrate) -> syn::Path {
+    krate.get_path()
+}
+
+/// Determines the crate paths to use by looking at the [metadata][cargo_metadata] of the project.
 ///
-/// This process is a bit hacky, we run `cargo metadata` internally which
-/// resolves the current package but creates a new `Cargo.lock` file in the
-/// process. This is not a problem for regular workspaces but becomes an issue
-/// during publishing with `cargo publish` if the project does not ignore
-/// `Cargo.lock` in `.gitignore`, because then cargo can't proceed with
-/// publishing the crate because the created `Cargo.lock` leads to a modified
-/// workspace, not the `CARGO_MANIFEST_DIR` but the workspace `cargo publish`
-/// created in `./target/package/..`. Therefore we check prior to executing
-/// `cargo metadata` if a `Cargo.lock` file exists and delete it afterwards if
-/// it was created by `cargo metadata`.
```

Comment:
what happened to all the docs? imo these provide additional useful context

Reply:
first paragraph is split up on the functions, i guess it's missing the explanation for why doing the entire thing (which is that we can have either ethers_core, ethers::core or both)

tbh second paragraph is excessive, deducible from the code (store .exist(), rm after)

Reply:
fine with removing